### ISSUE TITLE
Build a modular Laravel authorization foundation

### ISSUE THREAD

**Issue by lurzar** [lurzar/expenses-app#13]
## Summary

Implement the approved Laravel-native, database-backed authorization foundation so present and future modules can add roles, permissions, and policies through one secure convention.

## Current context

- Planning currently registers `PlanningPolicy` and its controllers call `Gate::authorize`, but authorization is limited to record ownership and there is no reusable role/permission persistence layer.
- `User` has no role/permission integration.
- The existing `superAdmin()` factory/seeder state creates demonstration data only; it is not a production role or provisioning path.
- #131 owns the architecture/package decision and module authorization contract.
- Parent release tracker: #130.

## Target outcome

Expenses App has a dedicated authorization/access-control boundary integrated with Laravel Gate and policies, initial protected roles, a code-owned permission catalog, safe assignment persistence, and a tested convention that new modules can extend without weakening record ownership.

## Analysis

- Product/value: new modules can declare capabilities and use existing roles without bespoke authorization infrastructure.
- Laravel/architecture: use the mechanism approved by #131 behind Laravel-native `can`, Gate, policy, middleware, and form-request APIs; keep controllers focused on HTTP orchestration.
- QA: test default denial, role inheritance, policy context, existing-user migration, cache invalidation, and every protected mutation.
- Security/privacy: role membership alone must not expose private Planning data; client capability flags are presentation only.
- Release/operations: migrations and permission synchronization must preserve existing users and support deterministic rollback/recovery.

## Scope

- Install and configure the authorization implementation approved by #131 using locked, audited dependencies.
- Own application-specific integration in a dedicated module/boundary consistent with the modular monolith.
- Add the required role, permission, and assignment schema with repository-compatible keys and constraints.
- Integrate the authenticated `User` model without exposing internal role/permission identifiers through Inertia.
- Define initial protected roles: `user`, `admin`, and `super-admin`.
- Implement an idempotent, code-owned permission catalog and synchronization service that future modules can extend.
- Register permissions through Laravel Gate and continue using model policies for contextual/ownership decisions.
- Update Planning authorization so the approved capability rule and existing owner-only invariant are both tested and preserved.
- Add explicit shared Inertia capability data only where needed for navigation/presentation; the server remains authoritative.
- Record assignment/catalog changes through the existing activity-log policy established by #131.
- Update architecture, interface, testing, and dependency documentation affected by the implementation.

## Out of scope

- Admin pages, role-editing forms, or user-management screens.
- Production super-admin provisioning/recovery workflow; tracked separately after this foundation.
- Direct permissions assigned to individual users.
- Arbitrary permission creation through a UI.
- Billing, subscriptions, tenant accounts, tenant-scoped roles, support impersonation, or access to private Planning data.

## Implementation plan

- [ ] Re-read and implement the approved #131 ADR and exact dependency decision.
- [ ] Add focused failing tests for schema, default denial, role inheritance, Gate integration, policy context, existing users, and permission-cache changes.
- [ ] Add/adapt migrations and configuration with safe constraints and rollback ordering.
- [ ] Integrate `User` and the module-owned permission catalog/synchronizer.
- [ ] Define protected system roles and seed/synchronize only deterministic non-secret authorization metadata.
- [ ] Register Laravel authorization and update protected routes/controllers/form requests/policies where required.
- [ ] Preserve and regress Planning owner-only access for view/delete/Expenses projection paths.
- [ ] Share minimal explicit capability props with Inertia and align TypeScript types.
- [ ] Add activity events without credentials, private financial values, or raw sensitive diagnostics.
- [ ] Update canonical docs and dependency records.
- [ ] Run focused tests, security diff review, migration/rollback validation, audits, and full checks.

## Dependencies and sequencing

- Parent: #130.
- Blocked by: #131.
- Blocks: super-admin lifecycle, Admin shell, and all Admin role-management issues in v2.2.0.
- Release target: `v2.2.0`.
- Planned base: `v2.2.0` after its separately authorized creation from exact `v2.x`.

## Acceptance criteria

- [ ] The implementation matches the approved #131 contract and uses a currently compatible locked dependency or approved local alternative.
- [ ] Existing users survive migration and receive only the approved default role through deterministic, tested behavior.
- [ ] `user`, `admin`, and `super-admin` roles exist with protected identifiers and documented purposes.
- [ ] Permissions are module/code-owned, uniquely named, idempotently synchronized, and available through Laravel Gate.
- [ ] Application enforcement uses Laravel `can`, policies, middleware, controller authorization, or form-request authorization rather than scattered role-name checks.
- [ ] Planning ownership and private Inertia payload boundaries remain intact for every role unless a later explicit permission/policy issue approves otherwise.
- [ ] Missing roles/permissions and stale or malformed input fail closed.
- [ ] Permission/assignment changes invalidate relevant caches and create minimized activity events.
- [ ] Migration rollback/recovery is tested on representative existing users without deleting user or Planning data.
- [ ] Frontend capabilities improve presentation only; direct unauthorized requests still receive a safe denial.
- [ ] Canonical architecture, interface, dependency, testing, and authorization documentation are current.

## Validation plan

- Focused Pest authorization tests covering allow/deny matrices, ownership, role inheritance, cache invalidation, and migration/rollback.
- Existing Planning and Expenses authorization regression suites.
- Representative fresh and upgrade migration/rollback against isolated SQLite plus PostgreSQL-specific SQL/runtime validation where required.
- `composer check` - exact-head repository gate.
- `git diff --check` - no whitespace errors.
- Locked Composer/npm audits and independent security diff review.

## Risks and rollback

- Risk: privilege escalation, lockout, leaked internal authorization data, stale cached access, or loss of assignments during rollback.
- Blast radius: authentication, every protected module, user schema, route access, Inertia props, and operational recovery.
- Rollback: back up assignment tables, reverse application usage before schema rollback, clear authorization cache, and retain user/Planning records. Do not automatically delete unknown production roles or assignments.

## Branching and PR

- Base: `v2.2.0` after release-branch authorization.
- Branch: `feat/13-modular-authorization-foundation`.
- PR: `Closes #13`; `Refs #130`; blocked by #131.

**Comment by lurzar** [comment]
Implementation is ready for review in #137.

- Base: `v2.2.0`
- Head: `feat/13-modular-authorization-foundation`
- Validated head: `f8203a24d5c85e64ca86bc0addcc4162b5df8813`
- Full `composer check`: passed
- Independent code review: Ready, no remaining findings

The PR body contains `Closes #13`. Because it targets the non-default `v2.2.0` integration branch, GitHub does not register an automatic closing reference; close this issue manually only after the PR is reviewed and merged.

**Comment by lurzar** [comment]
Completed by merged PR #137. The reviewed tree was preserved exactly in squash commit b612bac1ce8a638df1f832c65220c5f921ac3e8e on v2.2.0; full validation and rollback limitations remain recorded in the PR.
